master/buildbot/data/builds.py

```diff
@@ -20,6 +20,28 @@
 
 class Db2DataMixin(object):
 
+    def _injectProperties(self, data, props, filters):
+        """
+        This method injects Build's properties according to property filters.
+
+        Property filter allow one to request the Builds DATA API like so::
+
+            * api/v2/builds?property=propKey1&property=propKey2 (returns Build's properties which match given keys)
+            * api/v2/builds?property=* (returns all Build's properties)
+            * api/v2/builds?propKey1&property=propKey2&limit=30 (filters combination)
+
+
+        :param data: The Build data as a dict
+        :param props: The Build's properties as a dict
+        :param filters: Desired properties keys as a list
+
+        """
+        # by default none properties are returned
+        if data and props and filters:
+            data['properties'] = (props
```

Comment:
side note: I try to avoid writing functions that have side effects and this is clearly one of those cases.

I'd appreciate if you update the patch so it does not do this kind of things, but it's OK if you do not :smile: